lib/webmachine.ml

```diff
@@ -669,11 +669,27 @@ module Make(IO:IO) = struct
 
     method v3h11 : (Code.status_code * Header.t * 'body) IO.t =
       self#d "v3h11";
-      failwith "NYI: v3h11"
+      let d = self#get_request_header "if-unmodified-since" in
+      match d with
+      | None -> self#v3i12
+      | Some d' ->
+         match (Util.Date.parse_rfc1123_date d') with
+         | None -> self#v3i12
+         | Some _ -> self#v3h12
 
     method v3h12 : (Code.status_code * Header.t * 'body) IO.t =
       self#d "v3h12";
-      failwith "NYI: v3h12"
+      try
+        let u_mod = self#get_request_header "if-unmodified-since" in
+        let l_mod = self#get_request_header "last-modified" in
```

Comment:
`Last-Modified` is a response header. This is should be calling the `last_modified` resource method to compare it against the value in the `"if-unmodified-since"` header.